account/payments.go

```diff
@@ -476,6 +476,17 @@ func (a *Service) sendPayment(paymentHash string, payReq *lnrpc.PayReq, sendRequ
 AddInvoice encapsulate a given amount and description in a payment request
 */
 func (a *Service) AddInvoice(invoiceRequest *data.AddInvoiceRequest) (paymentRequest string, lspFee int64, err error) {
+	return a.AddInvoiceWithValidity(invoiceRequest, InvoiceValidityShort)
+}
+
+type InvoiceValidity int
+
+const (
+	InvoiceValidityShort InvoiceValidity = 0
+	InvoiceValidityLong  InvoiceValidity = 1
+)
+
+func (a *Service) AddInvoiceWithValidity(invoiceRequest *data.AddInvoiceRequest, validity InvoiceValidity) (paymentRequest string, lspFee int64, err error) {
```

Comment:
I think instead of receiving LSPInfo and validity this method can take `OpeningFeeParams`.
This definition will serve  well the swap use case as only the `OpeningFeeParams` will be persisted insead of the whole LSPInfo.